### PR TITLE
Submit/fix vega lite

### PR DESCRIPTION
**Pull request by lammermann**
###### Motivation for this change
vega-lite binaries were not able to run, because they could not find the vega dependency. This change fixes this and adds a test to assure future changes will provide running executables.

###### Things done

- Built on platform(s)
  - [ x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/node-packages/default.nix [hunk 1]
```diff
@@ -387,18 +387,21 @@ let
...
-        # versions: https://github.com/vega/vega-lite/issues/6863.
         postInstall = ''
-          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2pdf \
-            --replace "npx -p vega vg2pdf"  "${self.vega-cli}/bin/vg2pdf"
-          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2svg \
-            --replace "npx -p vega vg2svg"  "${self.vega-cli}/bin/vg2svg"
-          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2png \
-            --replace "npx -p vega vg2png"  "${self.vega-cli}/bin/vg2png"
+          cd node_modules
+          vega_dependencies=`ls ${self.vega-cli}/lib/node_modules/vega-cli/node_modules`
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/default.nix:392]:
```suggestion
          vega_dependencies=$(ls ${self.vega-cli}/lib/node_modules/vega-cli/node_modules)
```

File: pkgs/development/node-packages/default.nix [hunk 2]
```diff
@@ -387,18 +387,21 @@ let
...
+          for dep in $vega_dependencies
+          do
+            if [ ! -d "$dep" ]; then
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/default.nix:395]:
```suggestion
          for dep in $vega_dependencies; do
            if [[ ! -d $dep ]]; then
```

File: pkgs/development/node-packages/default.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/development/node-packages/default.nix]: is it better to use glob instead?

**Comment by lammermann** [comment]
Sorry for my ignorance. I've never done a pull request before. Do I have to to anything to apply the suggested changes or are they added to the pull request automatically?

**Comment by veprbl** [comment]
Feel free to squash them, or we could squash all commits on merge

**Comment by lammermann** [comment]
Ok I squashed them. Hope I did it right.

File: pkgs/development/node-packages/default.nix [hunk 3]
```diff
@@ -387,18 +387,20 @@ let
...
         postInstall = ''
-          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2pdf \
-            --replace "npx -p vega vg2pdf"  "${self.vega-cli}/bin/vg2pdf"
-          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2svg \
-            --replace "npx -p vega vg2svg"  "${self.vega-cli}/bin/vg2svg"
-          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2png \
-            --replace "npx -p vega vg2png"  "${self.vega-cli}/bin/vg2png"
+          cd node_modules
+          vega_dependencies=$(ls ${self.vega-cli}/lib/node_modules/vega-cli/node_modules)
+          for dep in $vega_dependencies; do
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/default.nix:393]:
```suggestion
          for dep in ${self.vega-cli}/lib/node_modules/vega-cli/node_modules/*; do
```

**Pull request merged**